### PR TITLE
Improve logical expression

### PR DESCRIPTION
**Pull request by NahuFigueroa97**
|Related issue| Documentation| Testing issue|
|---|---|---|
| #16677 |[Link](https://docs.google.com/document/d/1N0p7pd7XszQwFU9HBjpB7txIw2DCcAsV-BKfEZSLixw/edit#bookmark=kix.6d9amw6d92gn) | Testing issue |

## Description

The new engine needs expand the logical expression capabilities, supporting varios data types, and comparators.

- [x]  Include all json data types.
        For `==` and ` !=` support all types of JSON data, numerical, string, array, object, boolean and null.
- [x]  For Operators `>` `<` `>=` `<=`
        Only support for numbers.
- [x]  Change in the build to make it support all operators. 
        `stageBuilderCheck` function.
- [x]  Identifies whether an operator is inside a helper function and takes it as a string.
- [x]  Improve the UT for stageBuilderCheck_test.cpp.
- [x]  Delete an extra file that is empty.

### CODE REVIEW

File: src/engine/test/source/builder/builders/stageBuilderCheck_test.cpp [hunk 1]
```diff
@@ -100,24 +100,151 @@ TEST(StageBuilderCheckTest, ListBuildsCorrectExpression)
...
+                    std::make_tuple(Json {R"("field<=null")"},
+                                    "Check stage: The \"<=\" operator only allows operate with numbers or string"),
+                    std::make_tuple(Json {R"("field>={\"key\":\"value\"}")"},
+                                    "Check stage: The \">=\" operator only allows operate with numbers or string"),
+                    std::make_tuple(Json {R"("field>=[\"value1\",\"value2\"]")"},
+                                    "Check stage: The \">=\" operator only allows operate with numbers or string"),
+                    std::make_tuple(Json {R"("field>=false")"},
+                                    "Check stage: The \">=\" operator only allows operate with numbers or string"),
+                    std::make_tuple(Json {R"("field>=null")"},
+                                    "Check stage: The \">=\" operator only allows operate with numbers or string")));
```
Comment (JorgeMarinoDev) [inline src/engine/test/source/builder/builders/stageBuilderCheck_test.cpp:222]: Are we sure Check stage all cases in Testing::values map to the same fmt string "`Check stage: The \">=\" operator only allows operate with numbers or string""`

File: src/engine/source/builder/builders/stageBuilderCheck.cpp [hunk 2]
```diff
@@ -103,39 +95,134 @@ base::Expression stageBuilderCheckExpression(const std::any& definition,
             field = term.substr(pos1 + 1, pos2 - pos1 - 1);
             value = term.substr(0, pos1) + term.substr(pos2);
```
Comment (JavierBejMen) [inline src/engine/source/builder/builders/stageBuilderCheck.cpp:96]: We don't set valueJson when is a helper function, this results in the operation builder generating a function to check if the field is null and not the desired function. Add tests for this use case and fix the implementation.

**Review by JavierBejMen — CHANGES_REQUESTED** [review]

File: src/engine/test/source/builder/builders/stageBuilderCheck_test.cpp (hunk 1 above)
Comment (JorgeMarinoDev) [reply src/engine/test/source/builder/builders/stageBuilderCheck_test.cpp:224]: Disregard. Clear now.

**Review by JorgeMarinoDev — APPROVED** [review]
LGTM

**Review by cborla — APPROVED** [review]
LGTM

**Review by juliancnn — APPROVED** [review]
LGTM!

**Pull request merged**